lib/pools.js

```diff
@@ -0,0 +1,23 @@
+export const pools = new Map();
+export const controller = new AbortController();
+
+export default {
+  get: (config, setup) => {
+    const key = JSON.stringify(config);
```

Comment:
We'll want to move the key derivation into a separate function, like keyOf or idOf.